tests/common.py

```diff
@@ -1304,11 +1304,12 @@ async def get_system_health_info(hass: HomeAssistant, domain: str) -> dict[str,
 @contextmanager
 def mock_config_flow(domain: str, config_flow: type[ConfigFlow]) -> None:
     """Mock a config flow handler."""
-    assert domain not in config_entries.HANDLERS
+    handler = config_entries.HANDLERS.get(domain)
     config_entries.HANDLERS[domain] = config_flow
     _LOGGER.info("Adding mock config flow: %s", domain)
     yield
-    config_entries.HANDLERS.pop(domain)
+    if handler:
```

Comment:
This is a weird change. It means that the mocked config flow is now impacting other tests too as it's no longer removed. Can't we just use

```python
with patch.dict(config_entries.HANDLERS, {domain: config_flow}):
    yield
```

Reply:
Opened https://github.com/home-assistant/core/pull/107357. Let's see if it works 👍